projects/schematics/src/add-spartacus/index.ts

```diff
@@ -0,0 +1,559 @@
+  const buffer = host.read(path);
+  if (!buffer) {
+    throw new SchematicsException(`Could not read file (${path}).`);
+  }
+  const content = buffer.toString();
+  const source = ts.createSourceFile(
+    path,
+    content,
+    ts.ScriptTarget.Latest,
+    true
+  );
+
+  return source;
+}
+
+function addImport(
+  host: Tree,
+  modulePath: string,
+  importText: string,
+  importPath: string
+) {
+  const moduleSource = getTsSourceFile(host, modulePath) as any;
+  if (!isImported(moduleSource, importText, importPath)) {
+    const change = insertImport(
+      moduleSource,
+      modulePath,
+      importText,
+      importPath
+    );
+    if (change) {
+      const recorder = host.beginUpdate(modulePath);
+      recorder.insertLeft(
+        (change as InsertChange).pos,
+        (change as InsertChange).toAdd
+      );
+      host.commitUpdate(recorder);
+    }
+  }
+}
+
+function importModule(host: Tree, modulePath: string, importText: string) {
+  const moduleSource = getTsSourceFile(host, modulePath);
+  const metadataChanges = addSymbolToNgModuleMetadata(
+    moduleSource,
+    modulePath,
+    'imports',
+    importText
+  );
+  if (metadataChanges) {
+    const recorder = host.beginUpdate(modulePath);
+    metadataChanges.forEach((change: InsertChange) => {
+      recorder.insertRight(change.pos, change.toAdd);
+    });
+    host.commitUpdate(recorder);
+  }
+}
+
+function getStorefrontConfig(options: SpartacusOptions): string {
+  const baseUrl = options.baseUrl || 'https://localhost:9002';
+  const baseUrlPart = `
+          baseUrl: '${baseUrl}',`;
+  const baseSite = options.baseSite || 'electronics';
+  return `{
+      backend: {
+        occ: {${options.useMetaTags ? '' : baseUrlPart}
+          prefix: '/rest/v2/',
+          legacy: false
+        }
+      },
+      authentication: {
+        client_id: 'mobile_android',
+          client_secret: 'secret'
+      },
+      context: {
+        baseSite: ['${baseSite}']
+      },
+      i18n: {
+        resources: translations,
+          chunks: translationChunksConfig,
+          fallbackLang: 'en'
+      }
+    }`;
+}
+
+function getLineFromTSFile(
+  host: Tree,
+  modulePath: string,
+  position: number
+): [number, number] {
+  const tsFile = getTsSourceFile(host, modulePath);
+
+  const lac = tsFile.getLineAndCharacterOfPosition(position);
+  const lineStart = tsFile.getPositionOfLineAndCharacter(lac.line, 0);
+  const nextLineStart = tsFile.getPositionOfLineAndCharacter(lac.line + 1, 0);
+
+  return [lineStart, nextLineStart - lineStart];
+}
+
+function removeServiceWorkerSetup(host: Tree, modulePath: string) {
+  const buffer = host.read(modulePath);
+
+  if (!buffer) {
+    return;
+  }
+
+  let fileContent = buffer.toString();
+
+  const serviceWorkerImport = `import { ServiceWorkerModule } from '@angular/service-worker';`;
+  const serviceWorkerModuleImport = `ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production })`;
```

Comment:
It's not the customer custom code, but default result of angular/pwa schematic, that we need to clean up (because we initialized service worker by ourselves, in spartacus libs)

Reply:
👍